### PR TITLE
feat: evm adapter

### PR DESCRIPTION
**Pull request by ibrica**

### CODE REVIEW

File: packages/evm-adapter/yarn-error.log [hunk 1]
```diff
@@ -0,0 +1,3807 @@
+Arguments:
```
Comment (martinavagyan) [inline packages/evm-adapter/yarn-error.log:1]: this has to be ignored

File: packages/evm-adapter/hardhat.config.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  networks: {
+    hardhat: {
+      accounts: {
+        accountsBalance: '10000000000000000000000',
+      },
+      chainId: 7545,
+    },
+    matic: {
+      url: 'https://rpc-mumbai.maticvigil.com',
+      accounts: [process.env.PRIVATE_KEY as string],
```
Comment (martinavagyan) [inline packages/evm-adapter/hardhat.config.ts:20]: String(process.env.PRIVATE_KEY) over `as`

File: packages/evm-adapter/package.json [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    "eslint-plugin-import": "^2.26.0",
+    "eslint-plugin-jest": "^26.1.4",
+    "eslint-plugin-prettier": "^4.0.0",
+    "hardhat": "^2.12.4",
+    "hardhat-gas-reporter": "^1.0.9",
+    "mocha": "^9.2.0",
+    "prettier": "^2.5.1",
+    "prettier-plugin-solidity": "^1.0.0",
+    "solidity-coverage": "^0.8.2",
+    "ts-mocha": "^9.0.2",
```
Comment (martinavagyan) [inline packages/evm-adapter/package.json:67]: do you need mocha ? can't we do everything with jest ?

File: packages/evm-adapter/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{
+  "exclude": ["dist", "test"],
+  "include": ["."],
+  "compilerOptions": {
+    "target": "ES2020" /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */,
```
Comment (martinavagyan) [inline packages/evm-adapter/tsconfig.json:5]: You can remove the comments, I think it's cleaner that way

File: packages/evm-adapter/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (martinavagyan) [inline packages/evm-adapter/tsconfig.json:1]: why don't we use the same tsconfig as for solana adapter ?

File: packages/evm-adapter/package.json [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  },
+  "homepage": "https://github.com/heliofi/solidity-adapter",
+  "dependencies": {
+    "ethers": "^5.7.2"
+  },
+  "author": "Ivan Britvic<ivbritvic@gmail.com>",
+  "devDependencies": {
+    "@ethersproject/abi": "^5.4.7",
+    "@ethersproject/providers": "^5.4.7",
+    "@heliofi/eslint-config-helio": "^0.0.4",
```
Comment (martinavagyan) [inline packages/evm-adapter/package.json:40]: this is a very old version can you please use the new one

File: packages/evm-adapter/.prettierrc [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (martinavagyan) [inline packages/evm-adapter/.prettierrc:1]: you won't need this config if you use our prettier plugin similar to solana adapter

Comment (martinavagyan) [reply packages/evm-adapter/.prettierrc:1]:
```JSON
"eslintConfig": {
    "plugins": [
      "prettier",
      "jest"
    ],
    "rules": {
      "prettier/prettier": "error"
    },
    "extends": [
      "@heliofi/eslint-config-helio",
      "prettier"
    ],
    "parserOptions": {
      "project": "./tsconfig.json"
    }
  },
  "prettier": {
    "singleQuote": true
  },
  ```

File: packages/evm-adapter/yarn-error.log (hunk 1 above)
Comment (ibrica) [reply packages/evm-adapter/yarn-error.log:1]: OK, added to .gitignore

File: packages/evm-adapter/hardhat.config.ts (hunk 2 above)
Comment (ibrica) [reply packages/evm-adapter/hardhat.config.ts:20]: OK

File: packages/evm-adapter/package.json (hunk 3 above)
Comment (ibrica) [reply packages/evm-adapter/package.json:67]: We don't need mocha, removed

File: packages/evm-adapter/tsconfig.json (hunk 4 above)
Comment (ibrica) [reply packages/evm-adapter/tsconfig.json:5]: OK

File: packages/evm-adapter/package.json (hunk 6 above)
Comment (ibrica) [reply packages/evm-adapter/package.json:40]: Hm, you're right don't know why hardhat initialized old version, moving ethers to 5.7.0

File: packages/evm-adapter/.prettierrc (hunk 7 above)
Comment (ibrica) [reply packages/evm-adapter/.prettierrc:1]: Ok

File: packages/evm-adapter/tsconfig.json (hunk 5 above)
Comment (martinavagyan) [reply packages/evm-adapter/tsconfig.json:1]: I think we can update the other one as well

**Review by martinavagyan — APPROVED** [review]

**Pull request merged**